src/sys/system_v.rs

```diff
@@ -0,0 +1,385 @@
+use std::ptr::{null, null_mut};
+
+use crate::errno::Errno;
+use crate::Result;
+
+use libc::{self, c_int, c_short, c_void, key_t, shmid_ds, size_t};
+#[cfg(target_env = "gnu")]
+use libc::{semid_ds, seminfo};
+
+#[derive(Debug, Default, Clone, Copy)]
+/// Type used to transform a raw number to an octal permission, while performing a clamp to u9
+///
+/// # Example
+///
+/// ```
+/// # use nix::errno::Errno;
+/// # use nix::sys::system_v::Permissions;
+///
+/// # fn main() -> Result<(), Errno> {
+/// assert_eq!(Permissions::new(511)?.get_permission(), &(0o0777 as u16));
+/// assert_eq!(Permissions::new(512).expect_err("512 is bigger than what u9 can store"), Errno::E2BIG);
+/// # Ok(())
+/// # }
+/// ```
+pub struct Permissions {
```

Comment:
For the last 9 bits of those `flg` arguments, Nix already has a type for it,
   `nix::sys::stat::Mode`, so a new type is probably not necessary.

Reply:
Fixed, wasn't aware of this type. This makes it depend on the `fs` module but I'm assuming it's fine?

Reply:
>  This makes it depend on the fs module but I'm assuming it's fine?

Yeah, this is unfortunate, ideally, this `Mode` type should be available if `#[cfg(any(feature="fs", feature="system_v")]`, doing this will require some extra work, so I think it is ok to make `system_v` depend on `fs`.